pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDReport.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+
+public class CPDReport {
+    private final Iterator<Match> matches;
+    private final Map<String, Integer> numberOfTokensPerFile;
+
+    CPDReport(final Iterator<Match> matches, final Map<String, Integer> numberOfTokensPerFile) {
+        this.matches = matches;
+        this.numberOfTokensPerFile = numberOfTokensPerFile;
+    }
+
+    public Iterator<Match> getMatches() {
+        return matches;
+    }
+
+    public Map<String, Integer> getNumberOfTokensPerFile() {
+        return Collections.unmodifiableMap(numberOfTokensPerFile);
```

Comment:
The unmodifiable map can be created once in the constructor, then we have a simple getter here.